modules/packages/slackpkg

```diff
@@ -0,0 +1,188 @@
+                       [ "${VALUE}" != "${DEFARCH}" ] ; then
+                    fatal "Bad Input: Architecture ${VALUE} does not match ${DEFARCH}."
+                fi
+                ;;
+            PackageType*)
+                TYP="${VALUE}"
+                ;;
+            *)
+                fatal "Invalid input: '${INPUT}'."
+                ;;
+        esac
+    done
+}
+
+get_package_data() {
+    if [ -n "${NAM}" ] && [ -z "${PKG}" ] ; then
+        PKG="${NAM}"
+    fi
+    if [ -n "${PKG}" ] ; then
+        case "${PKG}" in
+            /*|./*) # File Path.  It's a file
+                TYP='file'
+                if [ ! -e "${PKG}" ] ; then
+                    fatal "No such file: '${PKG}'."
+                fi
+                ;;
+            http://*|ftp://|file://) # URL, it's a file.
+                TYP='file'
+                ;;
+            *)
+                TYP='repo'
+                ;;
+        esac
+        echo "PackageType=${TYP}"
+        if [ "${TYP}" = 'repo' ] ; then
+            if [ -z "`echo ${PKGVER} | grep -E '^[0-9]'`" ] ; then
+                if [ "${PKGNAM}" != "${PKGVER}" ] ;  then
+                    PKGNAM="${PKGNAM}-${PKGVER}"
+                fi
+                PKGVER=''
+            fi
+            # Returning version for PackageType=repo is not advised
+            # https://docs.cfengine.com/latest/reference-standard-library-package_modules.html
+            # if [ -n "${PKGVER}" ] ; then
+            #     echo "Version=${PKGVER}"
+            # fi
+            echo "Name="${PKG}
+        else
+          echo "Name="$(echo ${PKG} |sed -e 's/-[0-9].*$//' -e 's/[^/]+\(.*$\)/\1/' -e 's/.*\///')
+          echo ${PKG} |sed -e 's/.*\///' -e 's/.*-\([_a-z0-9.?]\+\)-\(.*\)-.*/Version=\1\'$'\nArchitecture=\\2/'
+        fi
+    fi
+}
+
+if [ "${CMD}" = 'supports-api-version' ] ; then
+    echo '1'
+    exit 0
+else
+    read_options
+fi
+
+case "${CMD}" in
+    get-package-data)
+        get_package_data
+        ;;
+    list-installed)
+        ls -l /var/log/packages/ |sed -e 's/.* \([^ ]\+\)-\([_a-w0-9.?]\+\)-\(.*\)-.*/Name=\1\'$'\nVersion=\\2\\nArchitecture=\\3/'
+        ;;
+    list-updates-local|list-updates)
+        slackpkg -dialog=off -batch=on upgrade-all | sed -r 's!^(.+)-(.+)-(i[0-9]86|noarch|x86_64|x86|fw|npmjs)-(.*)$!Name=\1\nVersion=\2\nArchitecture=\3!g' | grep -vE "Total\ package|Do\ you\ wish\ to|Looking\ for|Checking\ local|^$"
+        ;;
+    repo-install|file-install|remove)
+        process_inputs "${CMD}" "${INPUTLIST}"
+        case "${CMD}" in
+            repo-install)
+                slackpkg -dialog=off -batch=on -default_answer=yes install ${PKG} 2>/dev/null >/dev/null
+                ;;
+            file-install)
+                installpkg ${PKG} 2>/dev/null >/dev/null
+                ;;
+            remove)
+                slackpkg -dialog=off -batch=on -default_answer=yes remove ${PKG} 2>/dev/null >/dev/null
+                ;;
```

Comment:
I was playing around with install/remove of bsd-games either by repo or file and it seems install/remove works fine for repo but only install works for file.

```
"/tmp/bsd-games-2.13-x86_64-12.txz"
#"bsd-games"
#  policy => "present",
  policy => "absent",
  package_module => slackpkg;
```

Can you test this out and see what might be wrong? Or if you can even reproduce my results?

Reply:
You know I suppose you are right. What you have currently does conform to the docs so let's go with that. Intuitively I think it makes sense for day-to-day use. The case I was trying is more of a testing-centric case than an everyday use case. :+1: